src/similarity.rs

```diff
@@ -194,16 +194,36 @@ pub trait Similarity {
     /// the embeddings. If the vectors are unit vectors (e.g. by virtue of
     /// calling `normalize`), this is the cosine similarity. At most, `limit`
     /// results are returned.
-    fn similarity(&self, word: &str, limit: usize) -> Option<Vec<WordSimilarity>>;
+    fn word_similarity(&self, word: &str, limit: usize) -> Option<Vec<WordSimilarity>>;
+
+    /// Find words that are similar to the query embedding.
+    ///
+    /// The similarity between the query embedding and other embeddings is
+    /// defined by the dot product of the embeddings. If the vectors are unit
+    /// vectors (e.g. by virtue of calling `normalize`), this is the cosine
+    /// similarity. At most, `limit` results are returned.
+    fn embedding_similarity(
```

Comment:
I don't like overpacking traits with somewhat different functionality. Could you split this into two traits: `WordSimilarity` and `EmbeddingSimilarity`. We could consider folding `word_similarity_by` and `embedding_similarity_by` into these two respective traits.

Reply:
Did that, although I had to rename the `WordSimilarity` struct to `WordSimilarityResult`